File: functions/runtime/src/Perper.WebJobs.Extensions/Cache/StreamListener.cs
using System.Collections.Generic;
using Perper.WebJobs.Extensions.Model;

#pragma warning disable 8618
namespace Perper.WebJobs.Extensions.Cache
{
    [PerperData]
    public class StreamListener
    {
        public string AgentDelegate { get; set; }
        public string Stream { get; set; }
        public int Parameter { get; set; }
        public Dictionary<string, object?> Filter { get; set; }
        public bool Replay { get; set; }
        public bool LocalToData { get; set; }
    }
}

File: functions/python/perper/cache/stream_listener.py
from collections import OrderedDict
from pyignite import GenericObjectMeta
from pyignite.datatypes import String, IntObject, BoolObject, MapObject


class StreamListener(
    metaclass=GenericObjectMeta,
    type_name="StreamListener",
    schema=OrderedDict(
        [
            ("agentdelegate", String),
            ("stream", String),
            ("parameter", IntObject),
            ("filter", MapObject),
            ("replay", BoolObject),
            ("localtodata", BoolObject),
        ]
    ),
):
    pass
